_posts/2021-04-22-release-1.13.0.md

```diff
@@ -0,0 +1,374 @@
+
+```sql
+-- set up a catalog
+CREATE
+CATALOG hive_catalog WITH (
+'type' = 'hive'
+);
+USE
+CATALOG hive_catalog;
+
+-- or use temporary objects
+CREATE
+TEMPORARY TABLE users (
+user_id BIGINT,
+user_name STRING,
+user_level STRING,
+region STRING,
+PRIMARY KEY (user_id) NOT ENFORCED
+) WITH (
+'connector' = 'upsert-kafka',
+'topic' = 'users',
+'properties.bootstrap.servers' = '...',
+'key.format' = 'csv',
+'value.format' = 'avro'
+);
+
+-- set the execution mode for jobs
+SET
+execution.runtime-mode=streaming;
+
+-- set the sync/async mode for INSERT INTOs
+SET
+table.dml-sync=true;
+
+-- set the job's parallelism
+SET
+parallism.default=100;
+
+-- restore state from the specific savepoint path
+SET
+execution.savepoint.path=/tmp/flink-savepoints/savepoint-bb0dab;
+
+INSERT INTO user_stats
+SELECT user_id, COUNT(*)
+FROM users
+GROUP BY user_id;
+```
+
+---
+
+## PyFlink
+
+### State access in DataStream API 
+Stateful stream processing has always been a major differentiator for Apache Flink from other data
+processing systems. While many operations in a
+dataflow simply look at one individual event at a time (for example an event parser), some
+operations remember information across multiple events (for example window operators). These
+operations are called stateful.
+
+The 1.12.0 release was the release when we reintroduced a rearchitected Python DataStream API, but
+it is the access to State that comes in this release that fully opens up the full potential of
+Apache Flink to Python users.
+
+### Row-based operations in Table API 
+
+Every Flink release brings the Python Table API closer to a full feature parity with the Java API.
+In this release the support for row-based operations was added.
+
+
+## Other improvements
+
+**Exception histories in the Web UI**
+
+From now on, the Flink Web UI will present up to n last exceptions that caused a job to fail apart
+from the last one.
+
+**Better exception / failure-cause reporting for unsuccessful checkpoints**
+
+Starting from the release Flink will provide statistics for checkpoints which failed or were
+aborted. Prior versions were reporting metrics such as e.g. size of persisted data or trigger time
+only in case a checkpoint succeeded. This change makes it easier to figure out a reason for the
+failure without looking into logs.
+
+**Exactly-once JDBC sink**
+
+From 1.13, JDBC sink can guarantee exactly-once delivery for XA-compliant databases. That means no
+data loss or duplication can occur in case of failover. Double processing (by Flink operators) can
+still occur but the result will only be delivered once.
+
+The feature is available to DataStream programs via the new JdbcSink.exactlyOnceSink method (and
+JdbcXaSinkFunction directly). The target database must be XA-compatible.
+
+**Improved consistency of behavior between different SQL time functions**
+
+**Hive Query syntax compatibility**
+
+**General User-Defined Aggregate Function Support in PyFlink Table API**
```

Comment:
@dianfu @HuangXingBo Could you help me fill this in.

Also a comment from @morsapaes:
> Just a note to say that this was announced in Flink 1.12, so maybe it's important to mention what has improved from the initial implementation...?
https://flink.apache.org/news/2020/12/10/release-1.12.0.html#other-improvements-to-pyflink

Reply:
Sure~

Reply:
Done

Reply:
Thanks!